### PR TITLE
add feature gamepad-controll.

### PR DESCRIPTION
**Pull request by saronpasu**
I tryed create code, gamepad controll.
sorry, this still have bugs.

### CODE REVIEW

File: src/controller.js [hunk 1]
```diff
@@ -18,6 +18,9 @@ function Controller(game) {
     this.RMB = 2;
     this.callback = [null, null, null];
 
+    this.gamepad = navigator.getGamepads[0];
+    this.gamepadButtleMode = false; // toggle normal/battle gamepad keymap
```
Comment (TatriX) [inline src/controller.js]: Probably you mean "battle"?

File: src/controller.js [hunk 2]
```diff
@@ -996,6 +1150,30 @@ function Controller(game) {
...
+            }
+        }
+        for (var i = 0; i < this.gamepad.axes.length; i++) {
+            if (this.gamepad.axes[i] != 0) {
+                if(i != 4 && i != 5 && Math.abs(this.gamepad.axes[i]) >= 0.5) {
+                    if(this.gamepad.axes[i] < 0) {
+                        var minus = this.gamepadAxesMap[i].minus.bind(this);
+                        minus();
+                    } else {
+                        var plus = this.gamepadAxesMap[i].plus.bind(this);
```
Comment (TatriX) [inline src/controller.js]: Probably you want something like this:
```
this.gamepadAxesMap[i].plus.call(this)
```

File: src/controller.js [hunk 3]
```diff
@@ -996,6 +1150,30 @@ function Controller(game) {
...
+                        plus();
+                    }
+                }
+            }
+        }
+        var loop = this.updateGamepad.bind(this);
+        requestAnimationFrame(loop);
```
Comment (TatriX) [inline src/controller.js]: You should call this function from the `controller.update` method. Probably in the
```
if (now - lastTickUpdate > 500) {
```
block.

File: src/controller.js (hunk 1 above)
Comment (saronpasu) [inline src/controller.js]: fight combo hotkeys( 1,2,3,4,5 ) keymap to gamepad.
"standard" gamepad have only 11 buttons.
In order to arrange keys, buttons so scarce.
So, I decided to add 'normal mode' and 'battle mode' with toggle switch.

File: src/controller.js (hunk 2 above)
Comment (saronpasu) [inline src/controller.js]: oh! its so cool coding.

File: src/controller.js (hunk 1 above)
Comment (TatriX) [inline src/controller.js]: Yes, I get it. But `gamepadButtleMode` then should be called `gamepadBattleMode`

**Pull request closed without merging**